src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -857,6 +859,8 @@ def process_disk_or_snapshot_create_namespace(namespace):
     if namespace.source:
         try:
             namespace.source_blob_uri, namespace.source_disk, namespace.source_snapshot = _figure_out_storage_source(namespace.resource_group_name, namespace.source)  # pylint: disable=line-too-long
+            if not namespace.source_blob_uri and namespace.source_storage_account_id:
+                raise CLIError('usage error: --source_storage_account_id is only used with blob uri')
         except CloudError:
             raise CLIError("Incorrect '--source' usage: --source VHD_BLOB_URI | SNAPSHOT | DISK")
```

Comment:
This should be updated:
"Incorrect '--source' usage: --source {SNAPSHOT | DISK} | --source VHD_BLOB_URI [--source-storage-account-id ID]") 

Also the error on line 863 should reference the same usage error string.